elb/src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -0,0 +1,649 @@
+            Set<String> addedSubnets = Sets.difference(subnets, oldSubnets);
+            if (!addedSubnets.isEmpty()) {
+                elbApi.getSubnetApi().attachLoadBalancerToSubnets(elbName, addedSubnets);
+            }
+            if (!removedSubnets.isEmpty()) {
+                elbApi.getSubnetApi().detachLoadBalancerFromSubnets(elbName, removedSubnets);
+            }
+
+        }
+    }
+
+    private void configureListeners(ELBApi elbApi, String elbName, LoadBalancer loadBalancer) {
+        Set<ListenerWithPolicies> listenerWithPolicies = loadBalancer.getListeners();
+        Set<Integer> listenerPorts = Sets.newHashSet();
+        for (ListenerWithPolicies listenerWithPolicy : listenerWithPolicies) {
+            listenerPorts.add(listenerWithPolicy.getPort());
+        }
+
+        if (!listenerPorts.isEmpty()) {
+            elbApi.getLoadBalancerApi().deleteLoadBalancerListeners(elbName, listenerPorts);
+        }
+        Listener listener = buildListener();
+        elbApi.getLoadBalancerApi().createLoadBalancerListeners(elbName, ImmutableSet.<Listener>of(listener));
+
+    }
+
+    private Listener buildListener() {
+        int loadBalancerPort = getRequiredConfig(LOAD_BALANCER_PORT);
+        int instancePort = getRequiredConfig(INSTANCE_PORT);
+        String instanceProtocol = getRequiredConfig(INSTANCE_PROTOCOL);
+        String loadBalancerProtocol = getRequiredConfig(LOAD_BALANCER_PROTOCOL);
+
+        return Listener.builder()
+                .protocol(Protocol.valueOf(loadBalancerProtocol))
+                .instancePort(instancePort)
+                .instanceProtocol(Protocol.valueOf(instanceProtocol))
+                .port(loadBalancerPort)
+                .build();
+    }
+
+    protected String generateUnusedElbName() {
+        int maxAttempts = 100;
+        String elbName = null;
+        ConfigBag setup = config().getBag();
+        for (int i = 0; i < maxAttempts; i++) {
+            elbName = new JcloudsMachineNamer().generateNewGroupId(setup);
+            boolean exists = doesLoadBalancerExist(elbName);
+            if (!exists) return elbName;
+            LOG.debug("Auto-generated ELB name {} in {} conflicts with existing; " +
+                    "trying again (attempt {}) to generate name", new Object[]{elbName, this, (i+2)});
+        }
+        throw new IllegalStateException(
+            "Failed to auto-generate unused ELB name after "+maxAttempts+" attempts (last attempt was "+elbName+")");
+    }
+
+    protected boolean doesLoadBalancerExist(String elbName) {
+        JcloudsLocation loc = getLocation();
+        String region = getRegionName(loc);
+
+        LoadBalancerServiceContext loadBalancerServiceContext = newLoadBalancerServiceContext(loc.getIdentity(), loc.getCredential());
+        try {
+            return loadBalancerServiceContext.getLoadBalancerService().getLoadBalancerMetadata(region+"/"+elbName) != null;
+        }
+        catch (Exception e){
+            return true;
```

Comment:
Worth logging the exception here. 
```
LOG.trace("Load balancer {} not found when checking existence", elbName, e);
``` 
And especially minor, but prefer `} catch {` and `} finally {`.